src/App.tsx

```diff
@@ -1,10 +1,62 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { getTodos } from './utils/todos';
 
-const USER_ID = 0;
+const USER_ID = '11161';
+
+enum SortCondition {
+  All,
+  Active,
+  Completed,
+}
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[] | null>(null);
+  const [isNotificationVisible, setIsNotificationVisible] = useState(true);
+  const [filterBy, setFilterBy] = useState(SortCondition.All);
+  const [visibleItemsCount, setVisibleItemsCount] = useState(0);
+
+  function setFilterCOndition(condition: SortCondition) {
+    setFilterBy(condition);
+  }
+
+  function showNotification() {
+    setIsNotificationVisible(true);
+
+    setTimeout(() => {
+      setIsNotificationVisible(false);
+    }, 3000);
+  }
+
+  function filterTodos(allTodos: Todo[]) {
+    let filteredTodos = allTodos;
+
+    if (filterBy === SortCondition.Active) {
+      filteredTodos = filteredTodos.filter(todo => !todo.completed);
+    } else if (filterBy === SortCondition.Completed) {
+      filteredTodos = filteredTodos.filter(todo => todo.completed);
+    }
+
+    setVisibleItemsCount(filteredTodos.length);
+    setTodos(filteredTodos);
+  }
+
+  useEffect(() => {
+    setIsNotificationVisible(false);
+
+    getTodos(USER_ID)
+      .then(allTodos => {
+        filterTodos(allTodos);
+      })
+      .catch(() => {
+        showNotification();
+      });
+  }, [filterBy]);
```

Comment:
why do you need to send a request to the server when the filter is changing?
better use `todos` state for all existing todos and filter them on the front, without extra requests. Always try to reduce the number of requests to a server when possible